state-chain/runtime/src/lib.rs

```diff
@@ -1615,6 +1615,33 @@ impl_runtime_apis! {
 					}
 				).collect::<Vec<_>>()
 		}
+
+		fn cf_boost_pools_depth() -> Vec<BoostPoolDepth> {
+
+			fn boost_pools_depth<I: 'static>() -> Vec<BoostPoolDepth>
+				where Runtime: pallet_cf_ingress_egress::Config<I> {
+
+				pallet_cf_ingress_egress::BoostPools::<Runtime, I>::iter().map(|(asset, tier, pool)|
+
+					BoostPoolDepth {
+						asset: asset.into(),
+						tier: tier as u16,
+						available_amount: pool.get_available_amount().into()
+					}
+
+				).collect()
+			}
+
+			ForeignChain::iter().flat_map(|chain| {
+				match chain {
+					ForeignChain::Ethereum => boost_pools_depth::<EthereumInstance>(),
+					ForeignChain::Polkadot => boost_pools_depth::<PolkadotInstance>(),
+					ForeignChain::Bitcoin => boost_pools_depth::<BitcoinInstance>(),
+					ForeignChain::Arbitrum => boost_pools_depth::<ArbitrumInstance>(),
```

Comment:
Do we have the logic in place for boosting deposits and deploying boost liquidity on ethereum/polkadot/arbitrum?

Reply:
I realized that we will have to rely on this RPC method to detect which pools are functional on the protocol side since the `BoostPoolCreated` event is not a reliable option on testnets (only on mainnet). With this schema, we won't be able to differentiate between functional and non-functional pools on the testnets.

Reply:
The reason I have all four here is to make sure we don't forget to add any of these if we do decide to add boost pools for other chains (and when adding a brand new chain, we'd be forced to modify this too due to exhaustive matching). 

> I realized that we will have to rely on this RPC method to detect which pools are functional

OK, I didn't know that. My thinking was that this could be implicit: if there is a pool for some asset, then the asset is boostable. What do you mean by "non-functional" pool excactly? Actually, as it stands now, we are creating a boost pool for every asset. Can we not just put restrictions on the frontend for which pools can be accessed?